### PR TITLE
Copy travis scripts from ocean repo

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
Switches to using pre-generated sociomantictsunami/dlang docker image as base
instead of building compilers each time in place.

### CODE REVIEW

File: ci/askpass.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+set +x # No secret printing!
+exec printf "${OCEAN_D2_PASS}"
```
Comment (nemanja-boric-sociomantic) [inline ci/askpass.sh]: There are several `OCEAN` references.

File: ci/travis-after-success.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	git commit -a -m 'Auto-convert to D2'
+	# Create the new tag
+	d2tag=$TRAVIS_TAG+d2.auto
+	git tag -m "$TRAVIS_TAG auto-converted to D2, see https://github.com/sociomantic-tsunami/ocean/releases/tag/$TRAVIS_TAG" "$d2tag"
+	# Push (making sure the credentials are not leaked and using a helper
+	# to get the password)
+	set +x
+	git -c "credential.https://github.com.username=${OCEAN_D2_USER}" \
+		-c "core.askPass=./ci/askpass.sh" \
+		push "https://github.com/sociomantic-tsunami/ocean-d2.git" "$d2tag"
```
Comment (nemanja-boric-sociomantic) [inline ci/travis-after-success.sh]: we probably don't need this, right?

File: ci/askpass.sh (hunk 1 above)
Comment (mihails-strasuns-sociomantic) [reply ci/askpass.sh]: Yeah will grep / cleanup it in a moment

File: ci/travis-install.sh [hunk 3]
```diff
@@ -1,19 +1,5 @@
...
-# And pushing will also be very fast (done only if there are valid credentials)
-set +x # No peeking of password!
-if test -n "$DOCKER_PASSWORD" -a -z "$TRAVIS_TAG" # Don't push tags
-then
-    docker login -u"$DOCKER_USERNAME" -p"$DOCKER_PASSWORD"
-    set -x
-    docker push "$img"
-fi
+# Build the docker image.
+docker build --pull -t ocean .
```
Comment (nemanja-boric-sociomantic) [inline ci/travis-install.sh]: swarm

File: ci/travis-script.sh [hunk 4]
```diff
@@ -1,7 +1,5 @@
 #!/bin/sh
 set -xe
 
-img=$(ci/travis-image-name.sh)
-
 # Run the actual tests
-docker run -ti --rm -v "$PWD:$PWD" -w "$PWD" -u "$(id -u)" "$img"  ci/travis.sh
+docker run -ti --rm -v "$PWD:$PWD" -w "$PWD" -u "$(id -u)" ocean ci/travis.sh
```
Comment (nemanja-boric-sociomantic) [inline ci/travis-script.sh]: ocean

**Comment by mihails-strasuns-sociomantic** [comment]
Updated:
- remove all ocean references
- removed d2 conversion bits that are not relevant for swarm yet

**Comment by mihails-strasuns-sociomantic** [comment]
Green

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**